git_machete/cmd.py

```diff
@@ -1570,15 +1517,246 @@ def flush_caches(self) -> None:
+        ans = self.ask_if(f"Branch {bold(b)} is behind its remote counterpart {bold(rb)}.\n"
+                          f"Pull {bold(b)} (fast-forward only) from {bold(remote)}?" + pretty_choices('y', 'N', 'q', 'yq'),
+                          f"Branch {bold(b)} is behind its remote counterpart {bold(rb)}.\nPulling {bold(b)} (fast-forward only) from {bold(remote)}...")
+        if ans in ('y', 'yes', 'yq'):
+            self.__git.pull_ff_only(remote, rb)
+            if ans == 'yq':
+                raise StopTraversal
+            self.flush_caches()
+            print("")
+        elif ans in ('n', 'N', 'q', 'quit'):
+            raise StopTraversal
+
+    def __sync_before_creating_pr(self) -> None:
+
+        self.expect_at_least_one_managed_branch()
+        self.__empty_line_status = True
+
+        cb = self.__git.current_branch()
+        try:
+            self.expect_in_managed_branches(cb)
+        except MacheteException:
+            self.add(cb)
+
+        if self.__is_merged_to_upstream(cb):
+            return
+
+        s, remote = self.__git.get_strict_remote_sync_status(cb)
+        statuses_to_sync = (UNTRACKED, AHEAD_OF_REMOTE, DIVERGED_FROM_AND_NEWER_THAN_REMOTE)
+        needs_remote_sync = s in statuses_to_sync
+
+        if needs_remote_sync:
+            if s == AHEAD_OF_REMOTE:
+                self.__handle_ahead_state(cb, remote, abort_on_quit=False)
+            elif s == UNTRACKED:
+                self.__handle_untracked_state(cb)
+            elif s == DIVERGED_FROM_AND_NEWER_THAN_REMOTE:
+                self.__handle_diverged_and_newer_state(cb, remote, abort_on_quit=False)
+
+            self.__print_new_line(False)
+            self.status(warn_on_yellow_edges=True)
+
+        else:
+            if s == BEHIND_REMOTE:
+                warn(f"Branch {cb} is in <b>BEHIND_REMOTE</b> state.\nConsider using 'git pull'.\n")
+                self.__print_new_line(False)
+                ans = self.ask_if("Proceed with pull request creation?" + pretty_choices('y', 'N', 'q', 'yq'),
+                                  "Proceeding with pull request creation...")
+            elif s == DIVERGED_FROM_AND_OLDER_THAN_REMOTE:
+                warn(f"Branch {cb} is in <b>DIVERGED_FROM_AND_OLDER_THAN_REMOTE</b> state.\nConsider using 'git reset --keep'.\n")
+                self.__print_new_line(False)
+                ans = self.ask_if("Proceed with pull request creation?" + pretty_choices('y', 'N', 'q', 'yq'),
+                                  "Proceeding with pull request creation...")
+            elif s == NO_REMOTES:
+                raise MacheteException("Could not create pull request - there is no remote repository!")
+            else:
+                ans = 'y'  # only IN SYNC status is left
+
+            if ans in ('y', 'yes', 'yq'):
+                return
+            elif ans in ('N', 'n', 'q', 'quit'):
+                raise MacheteException('Pull request creation interrupted.')
```

Comment:
Maybe just `raise StopTraversal`, as in other places?

Reply:
Hmm🤔, here the exception is not connected to `traverse` plus there has to message to user that there is nothing left for git_machete to do.